nova/virt/vmwareapi/cluster_util.py

```diff
@@ -429,3 +430,26 @@ def update_cluster_das_vm_override(session, cluster, vm_ref, operation='add',
     config_spec.dasVmConfigSpec = [das_vm_spec]
 
     reconfigure_cluster(session, cluster, config_spec)
+
+
+def get_cluster_median_cpu_mhz(session, cluster) -> int:
+    """Get the median CPU frequency for all hosts in a cluster.
+
+    We need this for reservation, since flavors specify reservation in cores,
+    but reservation of a VMware VM's CPUs is specified in Hz. We calculate the
+    median to prevent outliers (esp. hosts that might have much _less_ CPUs for
+    some reason) from skewing the result.
+    """
+    hosts = session._call_method(vutil, "get_object_property", cluster,
+                                 "host")
+    if not hosts:
+        raise CpuReservationNoHostsError(cluster=cluster)
+    host_hz = [
+        session._call_method(vutil, "get_object_property", host,
+                             "hardware.cpuInfo.hz")
+        for host in hosts.ManagedObjectReference]
```

Comment:
Please don't make an API call per host in the cluster. There's a method to get properties for a list of morefs.

Reply:
obsoleted